server/lib/parties/party-service.ts

```diff
@@ -216,26 +220,33 @@ export default class PartyService {
     party.members.set(user.id, user)
     this.clientSocketsToPartyId.set(clientSockets, partyId)
 
-    this.publishToParty(partyId, {
+    this.publisher.publish(getPartyPath(party.id), {
       type: 'join',
       user,
     })
-
-    this.unsubscribeFromInvites(party, user)
     this.subscribeToParty(clientSockets, party)
   }
 
-  private unsubscribeFromInvites(party: PartyRecord, user: PartyUser) {
-    this.nydus.publish(getInvitesPath(party.id, user.id), { type: 'removeInvite' })
-    // TODO(2Pac): Remove the invite notification once the server-side notification system is in.
-    const userSockets = this.userSocketsManager.getById(user.id)
-    if (userSockets) {
-      userSockets.unsubscribe(getInvitesPath(party.id, user.id))
-    }
+  private async clearInviteNotification(partyId: string, user: PartyUser) {
+    this.notificationService
+      .retrieveNotifications({ userId: user.id, type: NotificationType.PartyInvite })
+      .then(
+        userInviteNotifications => {
+          const notification = userInviteNotifications.filter(n => n.data.partyId === partyId)[0]
+          if (notification) {
+            this.notificationService.clearById(user.id, notification.id).catch(err => {
```

Comment:
If you make the containing callback `async`, you can `await` this instead of nesting attached callbacks. Then you could add a single catch handler (after the above `then` instead of as the second parameter) that would handle all errors inside of here.

(I don't think the separate log messages are particularly useful here, given that we have the error attached and it will have a stack trace)

At some point (doesn't need to be fixed as part of this PR), we should also really pass loggers down to services like this, so that they're linked to a `request` and such. Pretty annoying here that these operations are associated with a particular request but we'll have no idea which